src/ipc/handler.c

```diff
@@ -796,14 +796,14 @@ static int ipc_comp_value(uint32_t header, uint32_t cmd)
 	trace_ipc("ipc: comp %d -> cmd %d", data.comp_id, data.cmd);
 
 	/* get the component */
-	comp_dev = ipc_get_comp(_ipc, data.comp_id);
+	comp_dev = ipc_get_comp(_ipc, ((struct sof_ipc_ctrl_data *)_ipc->comp_data)->comp_id);
 	if (comp_dev == NULL){
 		trace_ipc_error("ipc: comp %d not found", data.comp_id);
 		return -ENODEV;
 	}
 	
 	/* get component values */
-	ret = ipc_comp_cmd(comp_dev->cd, cmd, &data);
+	ret = ipc_comp_cmd(comp_dev->cd, cmd, _ipc->comp_data);
```

Comment:
@dabekjakub it seems like we dont really need the data variable at here right? can we just remove it?

Reply:
it is used later down the line in 
component.h:286 static inline int comp_cmd(struct comp_dev *dev, int cmd, void *data)

Reply:
I guess it must be removed in all the places to avoid stop-on-warning when using another compiler.

Reply:
There might be another issue, if there are var-size data indicated by the `(_ipc->comp_data).rhdr.hdr.size` > fixed `sizeof(data)`, there might be OOB here:
```c
	/* write component values to the outbox */
	mailbox_hostbox_write(0, &data, data.rhdr.hdr.size);

Reply:
May be relevant to other functions as well. Need to analyze it on Monday.

Reply:
Agreed, I think we need 
```
struct sof_ipc_ctrl_data *out_data = NULL;

/* get component values */
	ret = ipc_comp_cmd(comp_dev->cd, cmd, _ipc->comp_data, &out_data);
	if (ret < 0) {
		trace_ipc_error("ipc: comp %d cmd %u failed %d", data.comp_id,
				data.cmd, ret);
		return ret;
	}

	/* write component values to the outbox */
	mailbox_hostbox_write(0, out_data, out_data.rhdr.hdr.size);
        rfree(out_data);
```

Where out_data is allocated in the component but freed here.